ably/realtime/connection.py

```diff
@@ -272,6 +303,10 @@ async def on_protocol_message(self, msg):
                 self.__connected_future = None
             else:
                 log.warn('CONNECTED message received but connected_future not set')
+            self.__in_suspended_state = False
+            if self.__ttl_task:
+                self.__ttl_task.cancel()
+            self.__connection_details = msg['connectionDetails']
```

Comment:
I think we should be using `msg.get('connectionDetails')` here, also it might be worth making a data class for connection details so that it's typed?

Reply:
I looked into making this a data class, it doesn't seem suitable to be a data class.

Reply:
Could you explain why? Even if it's just a regular python class I think that would be better

Reply:
The connection details has other embedded fields, Making this a data class or python class will mean we will have to pass in these fields individually every time.
`ConnectionDetails(msg["connectionDetails"]["field 1"], msg["connectionDetails"]["field 2"], ..., msg["connectionDetails"]["field 8"])`
An alternative will be to make the class have only one attribute
`ConnectionDetails(msg["connectionDetails"])`. I don't think there's an advantage to that

Reply:
I think you can easily get around this by using a static factory method. We would only need to add fields to it once we start using them in the library so you could start off by only adding the `max_idle_interval` field for now.

Here's a simple example:
```python
@dataclass
class ConnectionDetails:
    max_idle_interval: int

    def __init__(self, max_idle_interval: int):
        self.max_idle_interval = max_idle_interval

    @staticmethod
    def from_json(json_dict: dict):
        return ConnectionDetails(json_dict.get('maxIdleInterval'))
```

Then you could just do:
```python
connection_details = ConnectionDetails.from_json(msg["connectionDetails"])
```